src/Common/Common.Api/Datasets/Dataset.cs

```diff
@@ -51,6 +51,32 @@ public class Dataset
             };
         }
 
+        public static implicit operator PowerBI.Api.V2.Models.Dataset(Dataset dataset)
+        {
+            if (dataset == null)
+            {
+                return null;
+            }
+
+            // We can assign all properties to dataset but some values prevent from creating push dataset, thus I didn't set them.
```

Comment:
Would it be better to make a separate object for creating pushing datasets then? I'm not a fan of commented out code.

Reply:
At this moment, this is the only scenario to use this conversion but I can do it as well. How about adding ConvertToPushDataset method to convert explicitly?

Reply:
You could make a method that takes a Dataset and converts it to a PushDataset. That sounds reasonable to me.

Reply:
Thanks, I do that.